pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -0,0 +1,370 @@
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client   client.Client
+	mcClient *mcfgClient.MachineconfigurationV1Client
+	scheme   *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a ComplianceRemediation object and makes changes based on the state read
+// and what is in the ComplianceRemediation.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileComplianceRemediation) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	var err error
+
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ComplianceRemediation")
+
+	// Fetch the ComplianceRemediation instance
+	remediationInstance := &complianceoperatorv1alpha1.ComplianceRemediation{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, remediationInstance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// First, we'll reconcile the MC that is the result of applying Remediations
+	switch remediationInstance.Spec.Type {
+	case complianceoperatorv1alpha1.McRemediation:
+		reqLogger.Info("Reconciling a MachineConfig remediation")
+		err = r.reconcileMcRemediation(remediationInstance, reqLogger)
+	default:
+		err = fmt.Errorf("unsupported remediation type %s", remediationInstance.Spec.Type)
+	}
+
+	// this would have been much nicer with go 1.13 using errors.Is()
+	if err != nil {
+		ccErr, ok := err.(*complianceCtrlError)
+		if ok {
+			if ccErr.canRetry {
+				reqLogger.Error(err, "Retrying error")
+				return reconcile.Result{}, err
+			}
+		}
+		reqLogger.Error(err, "Non-retriable error")
+		return reconcile.Result{}, nil
+	}
+
+	// Second, we'll reconcile the status of the Remediation itself
+	err = r.reconcileRemediationStatus(remediationInstance, reqLogger)
+	// this would have been much nicer with go 1.13 using errors.Is()
+	if err != nil {
+		ccErr, ok := err.(*complianceCtrlError)
+		if ok {
+			if ccErr.canRetry {
+				reqLogger.Error(err, "Retrying error")
+				return reconcile.Result{}, err
+			}
+		}
+		reqLogger.Error(err, "Non-retriable error")
+		return reconcile.Result{}, nil
+	}
+
+	reqLogger.Info("Done reconciling")
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileComplianceRemediation) reconcileMcRemediation(instance *complianceoperatorv1alpha1.ComplianceRemediation, logger logr.Logger) error {
```

Comment:
Please add some comments clarifying the action of de-selection of existing remediations

Reply:
Done in the comment above the function